src/parser/helper/Language.ts

```diff
@@ -102,6 +109,57 @@ export const fileExtensionToLanguage = new Map([
     ["sh", Language.Bash],
 ]);
 
+/**
+ * Maps supported file extensions to the corresponding programming languages.
+ * For case-sensitive file extensions.
+ */
+const caseSensitiveFileExtensionToLanguage = new Map([
+    ["c", Language.C],
+    ["C", Language.CPlusPlus],
+    ["H", Language.CPlusPlus], // lowercase .h has a special treatment
+]);
+
+/**
+ * Estimates the language of a file based upon the file extension and file path.
+ * @param filePath Path to the file, including the file extension.
+ * @param config Configuration to apply.
+ * @param pathModule ONLY FOR TESTING PURPOSES: overrides the platform-specific path module.
+ */
+export function assumeLanguageFromFilePath(
+    filePath: string,
+    config: Configuration,
+    pathModule = path,
+) {
+    const fileExtension: string = getFileExtension(filePath);
+
+    // Handling of the parse .h as C option:
+    if (fileExtension === "h") {
+        if (config.parseAllAsC) {
```

Comment:
```suggestion
        if (config.parseAllHeaderFilesAsC) {
```

Reply:
Well, not all, but only the once with the file extension `.h`. Thats not the only file extension for headers, there are also `.hpp`, `.hh`, `.hxx` or `.H` (probably even more).
I struggled there because I wanted to name it `parseAllHAsC` just like the command line option "parse-h-as-c", but that does really not work and is completely not comprehensible in camel case.

Reply:
I have talked with the others about it and they find `parseAllHAsC` acceptable. So I will go with that.